lib/delayed/backend/active_record.rb

```diff
@@ -138,23 +138,35 @@ def self.jobs_count(flavor,
         # or opts[:flavor] = <some flavor> to perform on all jobs of that flavor
         def self.bulk_update(action, opts)
           scope = if opts[:flavor]
-            raise("Can't bulk update failed jobs") if opts[:flavor].to_s == 'failed'
-            self.scope_for_flavor(opts[:flavor], opts[:query])
+            raise("Can't #{action.to_s} failed jobs") if opts[:flavor].to_s == 'failed' && action != 'destroy'
+            scopes = []
+            scopes << self.scope_for_flavor(opts[:flavor], opts[:query])
           elsif opts[:ids]
-            self.where(:id => opts[:ids])
+            scopes = []
+            scopes << Delayed::Job.where(:id => opts[:ids])
+            failed_jobs = Delayed::Job::Failed.where(:id => opts[:ids])
+            raise("Can't #{action.to_s} failed jobs") if failed_jobs.any? && action != 'destroy'
```

Comment:
Thinking about this more: I think this will run into the same problem as the change I made to `jobs_controller` where you might inadvertently act on both a failed and non-failed job with the same ID. I'm not sure of a good way to handle that. As written, this lets you bulk_update mixed types of jobs (e.g. failed-job1, failed-job2, regular-job3). I could change it so that `flavor` is required if you want to bulk update a bunch of failed jobs, and lose the mixed types.